gen.ml

```diff
@@ -1568,6 +1582,10 @@ module Restart = struct
 
   let take_while p e () = take_while p (e ())
 
+  let fold_while f s e =
+    let acc, gen = fold_while f s (e ()) in
+    acc, fun () -> gen
+
```

Comment:
Here we see that if `fold_while` were to return only the accumulator, its definition would be as simple as `Restart.fold`.